flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/table/MongoDynamicTableFactory.java

```diff
@@ -151,16 +146,11 @@ public DynamicTableSink createDynamicTableSink(Context context) {
         MongoConfiguration config = new MongoConfiguration(helper.getOptions());
         helper.validate();
 
-        ResolvedSchema schema = context.getCatalogTable().getResolvedSchema();
-        SerializableFunction<RowData, BsonValue> keyExtractor =
-                MongoKeyExtractor.createKeyExtractor(schema);
-
         return new MongoDynamicTableSink(
                 getConnectionOptions(config),
                 getWriteOptions(config),
                 config.getSinkParallelism(),
-                context.getPhysicalRowDataType(),
-                keyExtractor);
+                context.getCatalogTable().getResolvedSchema());
```

Comment:
Personally, I would keep out the catalog's schema from the sink. The sink has no schema but a data type. Let's use a boolean flag in the constructor `isUpsert`.

Reply:
OK